### DISCUSSION TITLE
Admin panel only?

### DISCUSSION THREAD

**Discussion by cheesegrits** [HollyIT/filament-static-assets#3] [General]
Is this specific to the admin panel, or does it help resolve issues with assets for standalone form builder usage?

Most of my Filament usage atm is "front end", and I've run in to a lot of frustrating issues where plugins leverage the admin panel plugin handler asset management, with no provision for form builder use outside of the admin panel.

**Comment by HollyIT** [comment]
Yeah, that's one of the big problems with the form package being split off. Static assets are handled by FilamentManager and the PluginServiceProviders, which are all part of the base admin package. I have no way to correct that in this package, as the actual registration of JS, CSS, and JSData all happens under the hood in the Filament base PluginServiceProvider class, when the ServingFilament event gets fired. 

This is where I think Filament really needs to rethink asset handling. I would love to see something along the lines of what Drupal did years ago with a "library" system for managing static dependencies. Of course, that would be a big, breaking change and have to take place in something like V3.
